### PR TITLE
test: Make pytest generate coverage report and sonar read it

### PR DESCRIPTION
**Pull request by yuryfirebolt**
The pipeline fails on the branch, because it depends on the changes, that should be in main

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -16,3 +16,10 @@ jobs:
     uses: firebolt-db/firebolt-python-sdk/.github/workflows/code-check.yml@main
   unit-tests:
     uses: firebolt-db/firebolt-python-sdk/.github/workflows/unit-tests.yml@main
+  security-scan:
+    needs: [unit-tests]
+    uses: firebolt-db/firebolt-python-sdk/.github/workflows/security-scan.yml@main
```
Comment (ptiurin) [inline .github/workflows/pull-request.yml:21]: you can use `security-scan.yml@<your_branch_name>` to test

**Review by ptiurin — COMMENTED** [review]
I don't like the coupling between security-scan and the rest of CI. Ideally, security-scan should be a plug and play as we expose it through the templates: https://github.com/firebolt-db/.github/blob/main/workflow-templates/security-scan-org.yml (setup doc: https://packboard.atlassian.net/wiki/spaces/PACB/pages/2813231305/New+Project+CI+setup).
I'm thinking it should stay a stand-alone workflow, with a step that calculates the coverage. This step would be a placeholder for projects to add their own coverage calculations (since we have other language projects apart from Python).

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (yuryfirebolt) [reply .github/workflows/pull-request.yml:21]: yes, did it for testing purposes in a separate branch, now want to push it to main

**Review by ptiurin — APPROVED** [review]
As discussed on the daily, this is a necessary step. We'll need to modify the templates after this change is confirmed to include this new dependency.

**Pull request merged**